### DISCUSSION TITLE
Støtte lesing fra MQTT (Tibber etc)

### DISCUSSION THREAD

**Discussion by toreamun** [toreamun/amshan-homeassistant#23] [Announcements]
Integrasjonen støtter nå lesing av data fra MQTT. Dette gjør at man kan benytte adaptere som publiserer til MQTT, deriblant [Tibber Pulse](https://tibber.com/no/store/produkt/pulse).

Se https://github.com/iotux/ElWiz#oppsett-av-pulse for konfigurering av MQTT for Tibber Pulse. Det er fint med tilbakemeldinger i diskusjonsgruppen og eventuelt som issues.

<hr /><em>This discussion was created from the release <a href='https://github.com/toreamun/amshan-homeassistant/releases/tag/2022.1.3-beta'>Støtte lesing fra MQTT (Tibber etc)</a>.</em>

**Comment by tfriberg** [comment]
In the mqtt_url field, valid information is `user:password@host`.
At least if you've set up an mqtt-user (which I believe you should)

**Reply by toreamun** [reply]
Yes, I updated the wiki https://github.com/toreamun/amshan-homeassistant/wiki/Lesere-MQTT#tibber-pulse last week.

**Comment by howi42** [comment]
I am using Node-Red to send Aidon data to Mosquitto ( Home Assistant ).
Checking Mosquitto, I see that the information is ok ( json formatted arrays ).
The AMSHAN integration seems to have problems with these arrays ( or the broker usage ).
AMSHAN asks for the topic to listen to,  but ends up running in a timeout.
Something I did wrong ?

**Comment by howi42** [comment]
I solved this problem, removed the JSON formatting

**Comment by parhedberg** [comment]
I am using EMQX as mqtt broker in my homeassistant. This broker has a feature called data bridge, that looks like it can forward incomming messages to another broker.

This would i my world make it possible to have both local access to Tibber pulse, and use the Tibber app, if we could know the right config for tibber mqtt.

Has anyone tried this?

**Reply by tfriberg** [reply]
I have tried following the steps from [LocalPulse2Tibber](https://github.com/MSkjel/LocalPulse2Tibber) where you copy all the data from the Tibber Pulse form (10.133.70.1) and save bridge.conf in /share/mosquitto/ and CA.ca, Cert.crt and Priv.key in /share/mosquitto

```
connection bridge-to-tibber
bridge_cafile /share/mosquitto/tibber_cert/CA.ca
bridge_certfile /share/mosquitto/tibber_cert/Cert.crt
bridge_keyfile /share/mosquitto/tibber_cert/Priv.key
address a1zhmn1192zl1a.iot.eu-west-1.amazonaws.com:8883
clientid tibber-pulse-[Pulse-ID]
try_private false
notifications false

# topic $aws/# out
# topic tibber-pulse-[Pulse-ID]/receive in

topic $aws/rules/push_to_kinesis/tibber-pulse-[Pulse-ID]/publish out
topic tibber-pulse-[Pulse-ID]/receive in
```

I have the Tibber Pulse badge in my Tibber app telling me it's properly configured, but I get no data, despite the bridge. 

The hunt continues.

**Reply by halsius** [reply]
I got that working fine (LocalPulse2Tibber instructions). What's not super clear in the guide is that you need to keep the topic exactly as-is in the tibber config page of pulse. And, after you've copied the certs from the config page you need to directly change to your real wifi password and save+apply. If you re-do the guide a new cert will be created and you have to copy it again.

I did set it up with certificate authentication to my local mqtt broker too, but I can't really see that changing auth to user/pass would break the messaging.

**Comment by howi42** [comment]
Why so complicated ?
As far as I know, Tibber does not really rely on the Pulse data, they get
their info from the power company.
That is why we can send the Pulse data just to us without creating trouble
to get the info, the power company itself has not API.
-Wolfgang

**Comment by parhedberg** [comment]
Well many of us uses Tibber pulse for real time loadbalancing while
charging a car for example.. This would not work while using Tibber pulse
only to local mqtt.

Den ons 22 mars 2023 13:18howi42 ***@***.***> skrev:

> Why so complicated ?
> As far as I know, Tibber does not really rely on the Pulse data, they get
> their info from the power company.
> That is why we can send the Pulse data just to us without creating trouble
> to get the info, the power company itself has not API.
> -Wolfgang
>

**Comment by howi42** [comment]
Not sure about that.
The same data I get locally via the Pulse, Tibber gets from the power
company.
If I am not missing anything here, I can do load balancing myself with the
Pulse data I have or leave it to Tibber to do this.
-Wolfgang

**Comment by jadesoturi** [comment]
Hi.

So a month or so ago, my Tibber Pulse that was setup through the Tibber integration in HA just became unavailable. Re-adding and re-paring the device resulted in just 1 entity being created. So I decided to try this integration since I'm no longer a tibber customer anyway.

Installed through HACS, reset Tibber, connected to wifi, added user:pass@ha-ip:1883 to the tibber configuration, along with the mqtt topics.

however, when i try to add the integration, I just get "timeout reading data" error after adding the topics(same as in the tibber config page pulse/publish and pulse/subscribe).

I installed mqtt-explorer and I am able to both read and write to pulse on these topics, but cant seem to add it to HA. 

Please advice:

HAOS, 
Core 2025.10.0b4
Supervisor 2025.09.3 
Mosquito Broker add-on.

//jadesoturi

**Reply by fikkaud** [reply]
Same issue here. Just spins and times out when entering topic on the dialogue: Subscribe to MQTT topic(s).
The broker receives data fine on the topic i enter
Message 1 received on tibber at 00:18:
b'\x08\xdc\xd9\x02\x12<\t\xac\xde\x86\x01\x00\x00\x00\x00\x12\x03HAN\x1a,~\xa0*
